### PR TITLE
Adding value_if_not_asked kwarg

### LINKED ISSUE #165: Provide optional kwarg to questions specifying a value if the question is not asked
I propose we add an optional kwarg to the questions (`BasicQuestion`, `Choice`, and `Confirm`) which allows users to specify a value if the question is not asked. I suggest this kwarg be named `value_if_not_asked` (but I'm open to other suggestions). Below, I propose *what* this new functionality might look like and *why* it's valuable.

# Current Functionality

Consider the following code:

```python
import columbo

interactions = [
    columbo.Confirm(
        "eats",
        "Do you eat to sustain life?"
    ),
    columbo.Confirm(
        "carnivore",
        "Do you eat meat?",
        should_ask=lambda answers: answers["eats"]
    )
]

columbo.get_answers(interactions)
```

If the user answers "No" to the `eats` question, the `carnivore` question will not be asked and the answers will *not* include a `'carnivore'` key:

```python
{'eats': False}
```

# Proposed Functionality

When implemented, my suggested `value_if_not_asked` kwarg would look like:

```python
import columbo

interactions = [
    columbo.Confirm(
        "eats",
        "Do you eat to sustain life?"
    ),
    columbo.Confirm(
        "carnivore",
        "Do you eat meat?",
        should_ask=lambda answers: answers["eats"],
        value_if_not_asked=False
    )
]

columbo.get_answers(interactions)
```

If the user answers "No" to the `eats` question, the `carnivore` question will not be asked *but...* the answers will include a `'carnivore'` key:

```python
{'eats': False, 'carnivore': False}
```

# Why?

This feature will be valuable because it allows users to guarantee that there is a key-value pair for each question (which is currently not the case as noted in the `Warning` [here](https://wayfair-incubator.github.io/columbo/latest/usage-guide/optional-questions-and-branching/#only-asking-some-questions)). There are many use-cases when it is very helpful to guarantee that there is a key-value pair for each question without having to either ask the user every question or write additional checks to see if a key is present before evaluating it (e.g. when using Columbo to populate cookiecutter templates as is being done [here](https://github.com/cookiecutter-flask/cookiecutter-flask)).

# Implementation Details

### What if the `value_if_not_asked` field is present without `should_ask`?

In this case, the question will be asked and the `value_if_not_asked` will be ignored.

### What are the type restrictions on the type of the `value_if_not_asked`?

The type of the `value_if_not_asked` will be the same as the type of answer and will be validated. So if `value_if_not_asked` is used for a `Confirm` question, it must be a boolean. If it is used for a `BasicQuestion`, it must be a string. If it is used for a `Choice`, it must be a string that is one of the options.

Any thoughts on this feature?

### PR DESCRIPTION
**Pull request by fhightower**
Fixes #165 .

### CODE REVIEW

**Comment by fhightower** [comment]
This is as far as I can get for now (I'll be AFK for a while). At first glance, I wasn't able to figure out why I was getting the [mypy error](https://github.com/wayfair-incubator/columbo/pull/169/checks?check_run_id=2898076821).

File: columbo/_interaction.py [hunk 1]
```diff
@@ -486,19 +546,21 @@ def get_answers(
...
+                result[interaction.name] = interaction.value_if_not_asked
         else:
             raise ValueError(f"Unsupported interaction type: {type(interaction)}")
 
     return result
 
 
 def canonical_arg_name(name: str) -> str:
-    sanizized_name = name.lower().replace(" ", "-").replace("_", "-").strip("-")
+    sanitized_name = name.lower().replace(" ", "-").replace("_", "-").strip("-")
```
Comment (fhightower) [inline columbo/_interaction.py:561]: Typo fix on a variable name

File: columbo/_interaction.py [hunk 2]
```diff
@@ -181,14 +189,22 @@ def __init__(
         :param cli_help: Optional help message to be displayed for command line interface.
         :param should_ask: If `None`, the question is asked. Otherwise, the callable will be passed the answers that
             have been provided this far and should return `True` if the question should be asked.
+        :param value_if_not_asked: If provided and if should_ask is being used, this value will be recorded as an answer if should_ask evaluates to False.
         """
-        super().__init__(name, message, cli_help, should_ask)
+        super().__init__(
+            name, message, cli_help, should_ask, value_if_not_asked=value_if_not_asked
```
Comment (jamescurtin) [inline columbo/_interaction.py:195]: Style nit: for consistency and readability, I think `cli_help` and `should_ask` should be used as kwargs as they are defined in the init function signature, just as you're doing with `value_if_not_asked`.

File: docs/examples/optional_questions_with_value_if_not_asked.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    columbo.BasicQuestion(
+        "dog_breed",
+        "What is the breed of the dog?",
+        should_ask=user_has_dog,
+        value_if_not_asked="n/a",
+        default="Basset Hound",
+    ),
+]
+
+user_answers = columbo.get_answers(interactions)
```
Comment (jamescurtin) [inline docs/examples/optional_questions_with_value_if_not_asked.py:26]: Super minor, but if the default on the `Confirm` was `False` and `no_user_input=True` was set on `get_answers`, this example would return the desired output without user interaction.

**Review by jamescurtin — APPROVED** [review]

File: docs/examples/optional_questions_with_value_if_not_asked.py (hunk 3 above)
Comment (fhightower) [reply docs/examples/optional_questions_with_value_if_not_asked.py:26]: @jamescurtin : I agree with what you said - Is that the wrong behavior? The behavior you described is what I expect, but happy to entertain other views.

Comment (jamescurtin) [reply docs/examples/optional_questions_with_value_if_not_asked.py:26]: Ah sorry--this thought wasn't clear. The current example is totally valid--my thought process was that switching the default value on the confirm would demonstrate the new functionality in the simplest way (_i.e._ if we wanted to be able to run the code examples from top to bottom and have it auto print out the expected result). But just sharing as a thought; no change requested

Comment (fhightower) [reply docs/examples/optional_questions_with_value_if_not_asked.py:26]: Ah, that makes sense. Thanks for the input! I'll consider update this.

**Review by plannigan — APPROVED** [review]

File: columbo/_interaction.py [hunk 4]
```diff
@@ -116,10 +117,17 @@ def __init__(
...
         :param should_ask: If `None`, the question is asked. Otherwise, the callable will be passed the answers that
             have been provided this far and should return `True` if the question should be asked.
+        :param value_if_not_asked: If provided and if should_ask is being used, this value will be recorded as an answer if should_ask evaluates to False.
         """
         self._name = name
         self._message = message
         self._cli_help = cli_help
+
+        if value_if_not_asked is not None and should_ask is None:
+            raise ValueError(
```
[the remark this answers is not in the document]
Comment (fhightower) [reply columbo/_interaction.py:127]: Great point - thanks for pointing this out. I'll get this updated.

**Pull request merged**